### ISSUE TITLE
Compiling openSSL to WebAssembly on Mac, the funtion RSA_generate_key_ex(rsa, 1024, bn, NULL) returns 0

### ISSUE THREAD

**Issue by sjjia** [openssl/openssl#18935]
#### C++ code:
```
#ifndef EM_PORT_API
#    if defined(__EMSCRIPTEN__)
#        include <emscripten.h>
#        if defined(__cplusplus)
#            define EM_PORT_API(rettype) extern "C" rettype EMSCRIPTEN_KEEPALIVE
#        else
#            define EM_PORT_API(rettype) rettype EMSCRIPTEN_KEEPALIVE
#        endif
#    else
#        if defined(__cplusplus)
#            define EM_PORT_API(rettype) extern "C" rettype
#        else
#            define EM_PORT_API(rettype) rettype
#        endif
#    endif
#endif


#include <openssl/rsa.h>
#include <openssl/x509.h>
#include <openssl/err.h>

EM_PORT_API (int) generateKeypair()
{
    RSA* rsa = RSA_new();
    BIGNUM *bn = BN_new();
    RSA* _keypair = NULL;
    
    int setRet = BN_set_word(bn, RSA_F4);
    int rsaRet = RSA_generate_key_ex(rsa, 1024, bn, NULL);
    printf("setRet = %d \n", setRet);
    printf("rsaRet = %d \n", rsaRet);
    if (setRet && rsaRet) {
        _keypair = rsa;
        printf("_keypair create success!! \n");
    }
    if (bn) {
        BN_free(bn);
    }
    if (rsa && _keypair==NULL) {
        RSA_free(rsa);
    }
    return 1;
}
```

#### apiTest.html code
```
<html>
  <head>
    <meta charset="utf-8">
    <title>Emscripten:Export1</title>
  </head>
  <body>
    <script>
    Module = {};
    Module.onRuntimeInitialized = function() {
      var ret = Module._generateKeypair();
    }
    </script>
    <script src="api.js"></script>
  </body>
</html>
```
#### Compiling openSSL to WebAssembly on Mac
the version is openssl-1.1.1d, download from https://www.openssl.org/source/old/1.1.1/openssl-1.1.1d.tar.gz
the Compiling step :

 * make clean
 * emConfigure ./Configure  darwin64-x86_64-cc -no-asm --api=1.1.0
 * Modify the MakeFile file:
     * CROSS_COMPILE=/Users/emsdk/upstream/emscripten/em -> CROSS_COMPILE=
     * CNF_CFLAGS=-arch x86_64 -> CNF_CFLAGS=
     * CFLAGS=-O3 -Wall -> CFLAGS=-O3 -Wall -s USE_PTHREADS -sSIDE_MODULE 
  * emmake make -j 12 build_generated libssl.a libcrypto.a

#### make api.js by emcc
```
emcc -sLLD_REPORT_UNDEFINED --no-entry -std=c++11 -s USE_PTHREADS -sMAIN_MODULE api.cpp openssl/libcrypto.a openssl/libssl.a  api.js
```

When running the demo，the funtion RSA_generate_key_ex(rsa, 1024, bn, NULL) returns 0.
![image](https://user-images.githubusercontent.com/39543586/182291961-8c763f4d-668c-4882-9254-8fabac8902ab.png)

I don't know which step is wrong ， Help Me!!!

**Comment by cx-github** [comment]
anyone know this question?

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of 3.4 dev barring further input
